### DISCUSSION TITLE
ESM Support

### DISCUSSION THREAD

**Discussion by hb20007** [tailwindlabs/tailwindcss#13434] [Ideas]
According to a [2022 comment by the creator of Tailwind CSS](https://github.com/tailwindlabs/tailwindcss/issues/8029#issuecomment-1086875656),

> Possible we'll support ESM in the future 👍🏻

Given that it is [recommended in the Next.js documentation](https://nodejs.org/api/packages.html#:~:text=This%20flag%20currently%20defaults%20to,where%20all%20sources%20are%20CommonJS.) to add `"type": "module"` to `package.json`, I believe that this is a good time to request this feature.

**Comment by adamwathan** [comment]
Hey! We added ESM support for the config file just over a year ago:

https://tailwindcss.com/blog/tailwindcss-v3-3#esm-and-type-script-support

On top of that, the Tailwind CSS v4 alpha ships as native ESM and CJS and will automatically install the right version for your project 👍

**Reply by hb20007** [reply]
Oh, I tried renaming my `tailwind.config.js` to `tailwind.config.mjs` and it wasn't recognized anymore, so I assumed that this is still unsupported.
